src/mscorlib/shared/System/Collections/HashHelpers.cs

```diff
@@ -33,6 +33,133 @@ internal static class HashHelpers
             187751, 225307, 270371, 324449, 389357, 467237, 560689, 672827, 807403, 968897, 1162687, 1395263,
             1674319, 2009191, 2411033, 2893249, 3471899, 4166287, 4999559, 5999471, 7199369};
 
+        public readonly struct PrimeInfo
+        {
+            public readonly int Prime;
+            public readonly uint Magic;
+            public readonly int Shift;
+
+            public PrimeInfo(int prime, uint magic, int shift)
+            {
+                Prime = prime;
+                Magic = magic;
+                Shift = shift;
+            }
+
+            public void Deconstruct(out int prime, out uint magic, out int shift)
+            {
+                prime = Prime;
+                magic = Magic;
+                shift = Shift;
+            }
+        }
+
+        // For more info see the book "Hacker's Delight" chapter 10.9 "Unsigned Division by Divisors >= 1"
+        // Generate & validate https://github.com/benaadams/Primes
+        public static readonly PrimeInfo[] PrimesInfo =
+        {
+            new PrimeInfo(3, 0x55555556, 0),
```

Comment:
This initialization pattern produces a lot of code bloat.

Reply:
Is there a better pattern?

I could do an array of int that proceeds in groups of 3 
e.g.
```
= { 3, 0x55555556, 0. 7, 0x92492493, 2, ...
```
But that seems a bit raw?

Reply:
Yes, it is a bit raw - we do it in number of other places.

Number of C# features force an unfortunate trade-off onto you: (pretty, big, slow) vs. (raw, small and fast).